CMakeLists.txt

```diff
@@ -0,0 +1,184 @@
+cmake_minimum_required(VERSION 3.1)
+
+project(opus LANGUAGES C)
+
+include(opus_config.cmake)
+include(opus_sources.cmake)
+include(opus_functions.cmake)
+
+add_library(opus ${opus_sources} ${opus_sources_float})
+
+set_target_properties(opus
+                      PROPERTIES SOVERSION
+                                 ${OPUS_LIBRARY_VERSION_MAJOR}
+                                 VERSION
+                                 ${OPUS_LIBRARY_VERSION})
+
+target_include_directories(opus
+                           PUBLIC $<BUILD_INTERFACE:
+                                  ${CMAKE_CURRENT_SOURCE_DIR}/include >)
+target_link_libraries(opus PRIVATE ${CMAKE_REQUIRED_LIBRARIES})
+
+target_include_directories(opus
+                           PUBLIC $<BUILD_INTERFACE:
+                                  ${CMAKE_CURRENT_SOURCE_DIR}/include >)
+target_link_libraries(opus PRIVATE ${CMAKE_REQUIRED_LIBRARIES})
+set_property(TARGET opus PROPERTY POSITION_INDEPENDENT_CODE ON)
+
+target_include_directories(opus
```

Comment:
`target_include_directories` with `PRIVATE` include dirs is not required here. You can set both `PRIVATE` and `PUBLIC` includes with one command.